app/src/main/java/com/example/peterchu/watplanner/home/HomeFragment.java

```diff
@@ -136,6 +146,7 @@ public void addCourses(List<Course> courses) {
     public void removeCourse(Course course) {
         courseAdapter.remove(course);
         courseAdapter.notifyDataSetChanged();
+        weekView.notifyDatasetChanged();
```

Comment:
It looks like you're telling the week view to refresh its data here to reflect modified data, but the line that actually modifies the week view's data is after this method is called in `onCourseRemoved`. Does this behave properly when testing?